cleverpush/src/main/java/com/cleverpush/banner/AppBannerModule.java

```diff
@@ -764,11 +844,79 @@ private void createBanners(Collection<Banner> banners) {
+          String value = bannerTargetEvent.getValue();
+          String fromValue = bannerTargetEvent.getFromValue();
+          String toValue = bannerTargetEvent.getToValue();
+
+          if (event == null || event.isEmpty() ||
+                  property == null || property.isEmpty() || Double.parseDouble(property) < 0 || Double.parseDouble(property) % 1 != 0 ||
+                  relationString == null || relationString.isEmpty() ||
+                  (relationString.equalsIgnoreCase("between") &&
+                          (fromValue == null || fromValue.isEmpty() || Double.parseDouble(fromValue) < 0 || Double.parseDouble(fromValue) % 1 != 0 ||
+                                  toValue == null || toValue.isEmpty() || Double.parseDouble(toValue) < 0 || Double.parseDouble(toValue) % 1 != 0)) ||
+                  (!relationString.equalsIgnoreCase("between") && (value == null || value.isEmpty() || Double.parseDouble(value) < 0 || Double.parseDouble(value) % 1 != 0))) {
+            Logger.d(TAG, "Skipping Target in banner " + banner.getId() + "because: Target values are not valid");
+            continue;
+          }
+
+          ArrayList<TableBannerTrackEvent> bannerTrackEvents = (ArrayList<TableBannerTrackEvent>) DatabaseClient.getInstance(CleverPush.context).
+                  getAppDatabase()
+                  .trackEventDao()
+                  .getBannerTrackEvent(banner.getId(), event, property, value != null ? value : "", String.valueOf(CheckFilterRelation.fromString(relationString)), fromValue != null ? fromValue : "", toValue != null ? toValue : "");
+
+          if (bannerTrackEvents.size() == 0) {
+            TableBannerTrackEvent task = new TableBannerTrackEvent();
+
+            task.setBannerId(banner.getId());
+            task.setTrackEventId(event);
+            task.setRelation(String.valueOf(CheckFilterRelation.fromString(relationString)));
+            task.setProperty(property);
+            task.setValue(value != null ? value : "");
+            task.setFromValue(fromValue != null ? fromValue : "");
+            task.setToValue(toValue != null ? toValue : "");
+            task.setCount(0);
+            task.setCreatedDateTime(getCleverPushInstance().getCurrentDateTime());
+            task.setUpdatedDateTime(getCleverPushInstance().getCurrentDateTime());
+
+            DatabaseClient.getInstance(CleverPush.context).getAppDatabase()
+                    .trackEventDao()
+                    .insert(task);
+          }
+          targetConditionTrue = this.checkTargetEventRelationFilter(CheckFilterRelation.fromString(relationString), property,
+                  value != null ? value : "", fromValue != null ? fromValue : "", toValue != null ? toValue : "", banner.getId(), event);
+
+          if (targetConditionTrue) {
+            targetEvents = true;
+            break;
+          }
+        }
+      }
 
+      if (isTriggerCondition && isTargetEvent) {
+        if (!targetEvents && !triggers) {
```

Comment:
should this be an || ?

Reply:
If I put `||` then both the trigger and target compulsory need to be satisfied otherwise banner will skip 
If I put `&&` then from the trigger and target any one's conditions satisfied banner will display 

@cpKevinrodriguesnobre What if a trigger has a different event and a target has a different event?

<img width="893" alt="Screenshot 2023-10-26 at 10 56 50 AM" src="https://github.com/cleverpush/cleverpush-android-sdk/assets/129381063/c74042ab-f96f-49b6-a279-fe68f4e46f
<img width="896" alt="Screenshot 2023-10-26 at 10 57 23 AM" src="https://github.com/cleverpush/cleverpush-android-sdk/assets/129381063/111a2a28-42a9-423f-92e0-6eb76c0e49b9">
33">

Reply:
@gbhrdt

Reply:
In my opinion, both should be satisfied, targeting and trigger. 
CC @cpandya25  (pls change for iOS) and @cpKevinrodriguesnobre

Reply:
I'll change && with ||. 
Now trigger event and target event both need to be satisfied to display the banner.